src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp

```diff
@@ -37,12 +37,7 @@ bool ShenandoahConcurrentRoots::should_do_concurrent_roots() {
 }
 
 bool ShenandoahConcurrentRoots::can_do_concurrent_class_unloading() {
-#if (defined(X86) || defined(AARCH64))
-  return ShenandoahCodeRootsStyle == 2 &&
-         ClassUnloading;
-#else
-  return false;
-#endif
+  return true;
```

Comment:
Hold on, this used to check `ClassUnloading`, should it continue to check it?

Reply:
No. I actually puzzled why checked it here, our concurrent roots scanner can handle both scenarios.
Maybe a leftover from concurrent roots to concurrent class unloading transition?

Reply:
It is maybe something that disables concurrent class unloading when `-ClassUnloading` is specified? Although it should have been handled in `shenandoahArguments.cpp`. Still, I'd prefer to leave the behavior exact, and then remove the `ClassUnloading` reference in a separate PR.

Reply:
Make sense, updated.